state_tracker.go

```diff
@@ -34,12 +35,15 @@ type SerializableState struct {
 	GhostferryVersion         string
 	LastKnownTableSchemaCache TableSchemaCache
 
-	LastSuccessfulPaginationKeys              map[string]uint64
-	CompletedTables                           map[string]bool
+	CompletedTables map[string]bool
+	BatchProgress   map[string]map[uint64]*BatchProgress
+
 	LastWrittenBinlogPosition                 mysql.Position
 	BinlogVerifyStore                         BinlogVerifySerializedStore
 	LastStoredBinlogPositionForInlineVerifier mysql.Position
 	LastStoredBinlogPositionForTargetVerifier mysql.Position
+
+	estimatedPaginationKeysPerSecond float64
```

Comment:
Do we know this is correct with the batching?

Reply:
I have indeed updated `EstimatedPaginationKeysPerSecond` but please review this part an extra time.